### PR TITLE
feat(core): add force running as massive action

### PR DESCRIPTION
**Pull request by stonebuzz**
add 'force running' as massive action

image

### CODE REVIEW

File: inc/task.class.php [hunk 1]
```diff
@@ -2019,6 +2024,21 @@ static function processMassiveActionsForOneItemtype(MassiveAction $ma, CommonDBT
...
 
+         case "forcestart":
+            foreach ($ids as $key) {
+               if ($pfTask->getFromDB($key)) {
+                  if ($pfTask->fields['is_active']) {
+                     $pfTask->forceRunning();
+                     $ma->itemDone($item->getType(), $key, MassiveAction::ACTION_OK);
+                  } else {
+                     $ma->itemDone($item->getType(), $key, MassiveAction::ACTION_KO);
+                     $ma->addMessage(sprintf(__('Task %1$s is disabled, force runnig aborted',
```
Comment (g-bougard) [inline inc/task.class.php:2035]: There a typo there: `runnig` ;-)

File: inc/task.class.php [hunk 2]
```diff
@@ -2019,6 +2024,21 @@ static function processMassiveActionsForOneItemtype(MassiveAction $ma, CommonDBT
...
             break;
 
+         case "forcestart":
+            foreach ($ids as $key) {
+               if ($pfTask->getFromDB($key)) {
+                  if ($pfTask->fields['is_active']) {
+                     $pfTask->forceRunning();
+                     $ma->itemDone($item->getType(), $key, MassiveAction::ACTION_OK);
+                  } else {
+                     $ma->itemDone($item->getType(), $key, MassiveAction::ACTION_KO);
```
Comment (cedric-anne) [inline inc/task.class.php:2034]: Not sure it should be considered as an error, maybe using `MassiveAction::NO_ACTION` result would be better.

**Comment by ddurieux** [comment]
We always refuse to add this feature with in massive action because if you do it on all computers, in case you have really many computers, you can 'ddos' the webserver.

**Review by ddurieux — CHANGES_REQUESTED** [review]
We always refuse to add this feature with in massive action because if you do it on all computers, in case you have really many computers, you can 'ddos' the webserver.

If you have a mechanism to prevent this, it could be possible to integrated it

**Comment by stonebuzz** [comment]
maybe queued inventory

https://github.com/fusioninventory/fusioninventory-for-glpi/pull/3066

**Comment by ddurieux** [comment]
no because it's not only inventory, it's about all requests of agent to server. When you force, there is prolog, inventory and all other tasks (deploy, netinventory, collect...)

**Comment by stonebuzz** [comment]
Even if all tasks are started, 
they should only run in the defined time slot?

It is the responsibility of the administrator to define the tasks correctly.

Currently the plugin already allows to force the start manually, if the user wants to run them all (one by one) he can also 'ddos' the server.

**Comment by ddurieux** [comment]
It's not a reason to report the fault to the administrator / technician...
We have the responsibility to guarantee and preserve the server.

It's not a reason someone want a feature we add it. In this case, if hr want it, hr must use tasks or code itself a script (it's his responsibility).

**Comment by ddurieux** [comment]
I close because not want add this code because introduce problems in the GLPI plugin / server

**Pull request closed without merging**